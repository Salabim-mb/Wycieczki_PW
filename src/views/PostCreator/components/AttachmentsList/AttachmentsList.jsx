import React from 'react';
import {
	List,
	ListItem,
	ListItemAvatar,
	ListItemText,
	IconButton,
	Avatar,
	ListItemSecondaryAction,
} from '@material-ui/core';
import { Delete, Attachment } from '@material-ui/icons';
import PropTypes from 'prop-types';

const AttachmentsList = ({ attachments, setAttachments }) => {
	const handleDeleteItem = id => {
		let newAttachments = [...attachments];
		const index = newAttachments.findIndex(attachment => attachment.id === id);
		newAttachments.splice(index, 1);
		newAttachments = newAttachments.map((attachment, i) => ({ ...attachment, id: i }));
		setAttachments(newAttachments);
	};

	return (
		<List>
			{attachments.map(({ file: { name }, id }) => (
				<ListItem key={id}>
					<ListItemAvatar>
						<Avatar>
							<Attachment />
						</Avatar>
					</ListItemAvatar>
					<ListItemText primary={name} />
					<ListItemSecondaryAction
						onClick={() => handleDeleteItem(id)}
						data-testid={`deleteIcon${id}`}
					>
						<IconButton edge="end" aria-label="delete">
							<Delete />
						</IconButton>
					</ListItemSecondaryAction>
				</ListItem>
			))}
		</List>
	);
};

AttachmentsList.propTypes = {
	attachments: PropTypes.arrayOf(
		PropTypes.shape({ id: PropTypes.number.isRequired, file: PropTypes.shape({}).isRequired }),
	).isRequired,
	setAttachments: PropTypes.func.isRequired,
};

export default AttachmentsList;
