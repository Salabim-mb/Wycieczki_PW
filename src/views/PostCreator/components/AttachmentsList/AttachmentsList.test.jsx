import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import AttachmentsList from './AttachmentsList';

describe('AttachmentsList', () => {
	let props;

	beforeEach(() => {
		props = {
			attachments: [{ id: 0, file: { name: 'attachment.jpg' } }],
			setAttachments: jest.fn(),
		};
	});

	it('should match snapshot', () => {
		const { container } = render(<AttachmentsList {...props} />);

		expect(container).toMatchSnapshot();
	});

	it('should delete item', () => {
		props.attachments = [
			{ id: 0, file: { name: 'attachment.jpg' } },
			{ id: 1, file: { name: 'attachment2.jpg' } },
			{ id: 2, file: { name: 'attachment3.jpg' } },
		];
		const { getByTestId } = render(<AttachmentsList {...props} />);

		fireEvent.click(getByTestId('deleteIcon1'));

		expect(props.setAttachments).toHaveBeenCalledWith([
			{ id: 0, file: { name: 'attachment.jpg' } },
			{ id: 1, file: { name: 'attachment3.jpg' } },
		]);
	});
});
