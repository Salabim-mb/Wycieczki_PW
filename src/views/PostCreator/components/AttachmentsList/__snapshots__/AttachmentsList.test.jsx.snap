// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttachmentsList should match snapshot 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-padding"
  >
    <li
      class="MuiListItem-container"
    >
      <div
        class="MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction"
      >
        <div
          class="MuiListItemAvatar-root"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M2 12.5C2 9.46 4.46 7 7.5 7H18c2.21 0 4 1.79 4 4s-1.79 4-4 4H9.5C8.12 15 7 13.88 7 12.5S8.12 10 9.5 10H17v2H9.41c-.55 0-.55 1 0 1H18c1.1 0 2-.9 2-2s-.9-2-2-2H7.5C5.57 9 4 10.57 4 12.5S5.57 16 7.5 16H17v2H7.5C4.46 18 2 15.54 2 12.5z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            attachment.jpg
          </span>
        </div>
      </div>
      <div
        class="MuiListItemSecondaryAction-root"
        data-testid="deleteIconabc"
      >
        <button
          aria-label="delete"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
  </ul>
</div>
`;
